shared/util.go

```diff
@@ -974,6 +974,35 @@ func ParseMetadata(metadata any) (map[string]any, error) {
 	return newMetadata, nil
 }
 
+// ParseJSONConfig extracts and parses a JSON-encoded configuration value from a config map.
+// It returns the parsed configuration as a map[string]string or an error if the key is missing or the JSON is malformed.
+func ParseJSONConfig(key string, configMap map[string]string) (map[string]string, error) {
```

Comment:
I dont think we need this very generic function in the shared (thus encouraging usage out of LXD) package.

Can we just do:

```go

	var parsed map[string]string
	err := json.Unmarshal([]byte(configMap[key]), &parsed)
	if err != nil {
		return nil, fmt.Errorf("Failed parsing %q: %w", key, err)
	}
```